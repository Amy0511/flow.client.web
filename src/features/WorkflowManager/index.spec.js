import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router";
import { WorkflowManagerContainer as WorkflowManager } from "./index";

jest.mock("Components/NavigateBack", () => "NavigateBack");
jest.mock("./Creator", () => "Creator");
jest.mock("./components/ActionBar", () => "ActionBar");

const mockfn = jest.fn();

const actions = {
  fetch: mockfn,
  reset: mockfn
};
const tasks = {
  isFetching: false,
  status: "success",
  error: "",
  data: []
};
const workflow = {
  isFetching: false,
  status: "success",
  error: "",
  data: []
};

describe("WorkflowManager --- Snapshot", () => {
  it("Capturing Snapshot of WorkflowManager", () => {
    const renderedValue = renderer
      .create(
        <MemoryRouter>
          <WorkflowManager
            tasks={tasks}
            workflow={workflow}
            tasksActions={actions}
            workflowRevisionActions={actions}
            workflowActions={actions}
          />
        </MemoryRouter>
      )
      .toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
});

describe("WorkflowManager --- Shallow render", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <MemoryRouter>
        <WorkflowManager
          tasks={tasks}
          workflow={workflow}
          tasksActions={actions}
          workflowActions={actions}
          workflowRevisionActions={actions}
        />
      </MemoryRouter>
    );
  });

  it("Render the DUMB component", () => {
    expect(wrapper.length).toEqual(1);
  });
});
