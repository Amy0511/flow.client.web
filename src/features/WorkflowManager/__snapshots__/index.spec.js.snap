// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkflowManager --- Snapshot Capturing Snapshot of WorkflowManager 1`] = `
<div
  className="c-workflow-designer"
/>
`;
